src/CommunityToolkit.Aspire.Hosting.Ollama/OllamaResourceBuilderExtensions.cs

```diff
@@ -55,63 +51,76 @@ public static IResourceBuilder<OllamaResource> WithDataVolume(this IResourceBuil
 
-        builder.Resource.AddModel(modelName);
-        return builder;
+        string resourceName = $"{builder.Resource.Name}-{modelName.Split(':')[0].Split('/').Last().Replace('.', '-')}";
+
+        return AddModel(builder, resourceName, modelName);
     }
 
     /// <summary>
-    /// Sets the default model to be configured on the Ollama server.
+    /// Adds a model to the Ollama container.
     /// </summary>
-    /// <param name="builder">The <see cref="IResourceBuilder{T}"/>.</param>
-    /// <param name="modelName">The name of the model.</param>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="modelName">The name of the LLM to download on initial startup.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<OllamaResource> WithDefaultModel(this IResourceBuilder<OllamaResource> builder, string modelName)
+    public static IResourceBuilder<OllamaModelResource> AddModel(this IResourceBuilder<OllamaResource> builder, string name, string modelName)
     {
         ArgumentNullException.ThrowIfNull(builder, nameof(builder));
         ArgumentException.ThrowIfNullOrWhiteSpace(modelName, nameof(modelName));
 
-        builder.Resource.SetDefaultModel(modelName);
-        return builder;
+        builder.ApplicationBuilder.Services.TryAddLifecycleHook<OllamaModelResourceLifecycleHook>();
+
+        builder.Resource.AddModel(modelName);
+        var modelResource = new OllamaModelResource(name, modelName, builder.Resource);
+
+        return builder.ApplicationBuilder.AddResource(modelResource);
     }
 
     /// <summary>
-    /// Adds a reference to an Ollama resource to the application model.
+    /// Adds a model from Hugging Face to the Ollama container.
     /// </summary>
-    /// <typeparam name="T">The type of the resource to add Ollama to.</typeparam>
-    /// <param name="builder">The <see cref="IResourceBuilder{T}"/>.</param>
-    /// <param name="ollama">The Ollama resource to add as a reference.</param>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="modelName">The name of the LLM from Hugging Face to download on initial startup.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    /// <remarks>
-    /// This method adds the connection string and model references to the application model.
-    /// </remarks>
-    public static IResourceBuilder<T> WithReference<T>(this IResourceBuilder<T> builder, IResourceBuilder<OllamaResource> ollama) where T : IResourceWithEnvironment
+    public static IResourceBuilder<OllamaModelResource> AddHuggingFaceModel(this IResourceBuilder<OllamaResource> builder, string name, string modelName)
```

Comment:
This method could do a pass-through to the `AddModel` with the prefix of `hf.co/`, which would reduce a little bit of code duplication.